### DISCUSSION TITLE
Slow controller.argument_value_resolver or first doctrine call

### DISCUSSION THREAD

**Discussion by henderkes** [symfony/symfony#50587] [Q&A] [answered]
Much like in #50531, although not as slow:

I'm facing a small performance problem that seemingly came out of nowhere, going back to older version of my code doesn't make a difference either. The first doctrine find takes an absurd amount of time (~150ms), while subsequent ones are still quick.

Testing code:

```php
#[Route(path: '/{id}/installation', name: 'seminar_installation')]
public function installationAction(Request $request, int $id, SeminarRepository $repo, Stopwatch $stopwatch)
{
    $stopwatch->start('first_find');
    $seminar = $repo->find(201038);
    $sevent1 = $stopwatch->stop('first_find');
    $stopwatch->start('second_find');
    $seminar = $repo->find($id);
    $sevent2 = $stopwatch->stop('second_find');
    $seminar->setInstallationNote(strval($sevent1->getDuration()));
    $seminar->setInstallationDisplayTitle(strval($sevent2->getDuration()));
    $form = $this->createForm(InstallationType::class, $seminar);
    $form->handleRequest($request);
    if ($form->isSubmitted() && $form->isValid()) {
        $this->getDoctrine()->getManager()->persist($seminar);
        $this->getDoctrine()->getManager()->flush();
        $this->addFlash('success', 'seminar.editInstallation');
    }
    return $this->render('seminar/installation.html.twig', [
        'form' => $form->createView(),
        'seminar' => $seminar,
    ]);
}
```

![image](https://github.com/symfony/symfony/assets/7896469/23d30b22-bf04-439a-82ce-36b81be49d3d)

161 ms for the first find operation, less than 10ms for anything after that. It would lead me to believe that It's connecting to the database when first called and keeps the connection up for later ones, but this is happening on both dev and staging environment and it hasn't been a thing a few months ago.

I'll hopefully be able to provide a blackfire profile soon.

**Comment by MatTheCat** [comment]
Does the issue still manifest when you have a fresh cache?

(BTW this is not a problem with arguments resolvers as they already ran before your controller’s code.)

**Reply by henderkes** [reply]
Yes, takes just as long after a fresh cache clear/warmup.

If I let the argument resolver figure out the Seminar entity, like you would typically do, it takes 150ms. That's why I mentioned it in the title.

**Comment by ghost** [comment]
AFAIR doctrine's `find()` caches the entity in UoW. First call will ask the database. Every next call with the same id will search for the entity inside UoW.

**Reply by henderkes** [reply]
First call is to a different entity, though, although it's most likely related to doctrines boot up. I'm just not sure what could've slowed it down to 150ms from the ~50ms it took a year ago.

Unfortunately it's been like this for a while and I only just noticed, the issue does not exist in a (very) old dev environment from over a year ago.

**Reply by ghost** [reply]
> First call is to a different entity

Sorry, I didn't notice that.

It looks like you will have to do some debugging to trace the line/method inside Doctrine's source code that slows down things.

**Comment by henderkes** [comment, marked as answer]
Setting doctrine caches to use redis (default before) brought me back to ~60ms initialisation time for doctrine. I will still run a profile when I get blackfire set up, though.
